bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +60,168 @@
     }
 
+    /**
+     * Format a description text to display on the thing main UI page
+     *
+     * @param delaySeconds the delay that will be added to the current time
+     * @return the description text
+     */
+    public String uiText(long delaySeconds) {
+        return "Waiting to make automatic reconnection attempt at " + LocalDateTime.now().plusSeconds(delaySeconds)
+                .truncatedTo(ChronoUnit.SECONDS).format(DateTimeFormatter.ofPattern("HH:MM:SS (cccc)"));
+    }
+
     /**
      * Initializes the GardenaSmart account.
+     * This method is called on a thread.
      */
-    private void initializeGardena() {
-        final GardenaAccountHandler instance = this;
-        scheduler.execute(() -> {
-            try {
-                GardenaConfig gardenaConfig = getThing().getConfiguration().as(GardenaConfig.class);
-                logger.debug("{}", gardenaConfig);
-
-                String id = getThing().getUID().getId();
-                gardenaSmart = new GardenaSmartImpl(id, gardenaConfig, instance, scheduler, httpClientFactory,
-                        webSocketFactory);
-                final GardenaDeviceDiscoveryService discoveryService = this.discoveryService;
-                if (discoveryService != null) {
-                    discoveryService.startScan(null);
-                    discoveryService.waitForScanFinishing();
-                }
-                updateStatus(ThingStatus.ONLINE);
-            } catch (GardenaException ex) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
-                disposeGardena();
-                if (ex.getStatus() == 429) {
-                    // if there was an error 429 (Too Many Requests), wait for 24 hours before trying again
-                    scheduleReinitialize(REINITIALIZE_DELAY_HOURS_LIMIT_EXCEEDED, TimeUnit.HOURS);
+    private synchronized void initializeGardena() {
+        try {
+            GardenaConfig gardenaConfig = getThing().getConfiguration().as(GardenaConfig.class);
+            logger.debug("{}", gardenaConfig);
+
+            String id = getThing().getUID().getId();
+            gardenaSmart = new GardenaSmartImpl(id, gardenaConfig, this, scheduler, httpClientFactory,
+                    webSocketFactory);
+            final GardenaDeviceDiscoveryService discoveryService = this.discoveryService;
+            if (discoveryService != null) {
+                discoveryService.startScan(null);
+                discoveryService.waitForScanFinishing();
+            }
+            reInitializationCausedBy429 = false;
+            updateStatus(ThingStatus.ONLINE);
+        } catch (GardenaException ex) {
+            logger.warn("{}", ex.getMessage());
+
+            long delaySecs;
+
+            synchronized (reInitializationCodeLock) {
+                Duration delay;
+                boolean isHttp429Error = (HttpStatus.TOO_MANY_REQUESTS_429 == ex.getStatus());
```

Comment:
I find `variable == constant` a bit more logical than `constant == variable`.

Reply:
Ok. Agreed.

( Its just that some people seem to prefer (say) `assertEquals(constant, variable)` or `constant.equals(variable)`. But I agree that `(variable == constant)` is better.. )

Reply:
Usually this is to eliminate risk of `NullPointerException` if variable might be null, e.g.:
```
"Hello".equals(a); // will never throw a NPE
a.equals("Hello"); // can throw an NPE.
```

Your `assertEquals()` example is not valid, as it takes _expected_ as first argument, so you can't choose the order.